blockchain/v2/routine.go

```diff
@@ -82,7 +98,21 @@ func (rt *Routine) start() {
 			return
 		}
 		rt.metrics.EventsOut.With("routine", rt.name).Add(1)
-		rt.logger.Debug(fmt.Sprintf("%s: produced %T %+v\n", rt.name, oEvent, oEvent))
+		rt.logger.Debug(fmt.Sprintf("%s: produced %T %+v", rt.name, oEvent, oEvent))
+
+		if e, ok := events[0].(Event); ok {
```

Comment:
I don't think we need to check this assertion with `ok`, since it was already asserted above at `oEvent, err := rt.handle(events[0].(Event))`, which would panic on failure.